libraries/cmake/source/thrift/CMakeLists.txt

```
@@ -95,9 +95,15 @@ function(thriftMain)
 
   # C++17 dropped support for random_shuffle. Add it back with a
   # custom header
-  target_compile_options(thirdparty_thrift PRIVATE
-    "${forced_include_file_flag}" "${CMAKE_CURRENT_SOURCE_DIR}/patches/random_shuffle.h"
-  )
+  if(DEFINED PLATFORM_WINDOWS)
+    target_compile_options(thirdparty_thrift PRIVATE
+      "${forced_include_file_flag}${CMAKE_CURRENT_SOURCE_DIR}/patches/random_shuffle.h"
+    )
+  else()
+    target_compile_options(thirdparty_thrift PRIVATE
+      "${forced_include_file_flag}" "${CMAKE_CURRENT_SOURCE_DIR}/patches/random_shuffle.h"
```

Might want to change this too, otherwise CMake can separate them on the command line as it's happening on Windows.
```suggestion
      "${forced_include_file_flag} ${CMAKE_CURRENT_SOURCE_DIR}/patches/random_shuffle.h"
```

---

I did try that, but then it fails on macOS :(